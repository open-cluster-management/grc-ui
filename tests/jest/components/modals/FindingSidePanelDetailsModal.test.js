/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import FindingSidePanelDetailsModal from '../../../../src-web/components/modals/FindingSidePanelDetailsModal'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom'
import * as reducers from '../../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { findingsSidePanelData, resourceType } from './ModalsTestingData'

describe('FindingSidePanelDetailsModal', () => {
  it('renders as expected', () => {
    const preloadedState = window.__PRELOADED_STATE__
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), preloadedState, composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const location = {
      'href': 'https://localhost:3000/multicloud/policies/findings',
      'ancestorOrigins': {},
      'origin': 'https://localhost:3000',
      'protocol': 'https:',
      'host': 'localhost:3000',
      'hostname': 'localhost',
      'port': '3000',
      'pathname': '/multicloud/policies/findings',
      'search': '',
      'hash': ''
    }
    const history = {}
    const updateModal = jest.fn()
    const component = shallow(
      <BrowserRouter>
        <FindingSidePanelDetailsModal
          data={findingsSidePanelData}
          title={'policy that is not compliant'}
          location={location}
          history={history}
          open={true}
          updateModal={updateModal}
          store={store}
          resourceType={resourceType}
          locale={'en'}
        />
      </BrowserRouter>
    )
    expect(component.instance().render()).toMatchSnapshot()
    expect(component.dive()).toMatchSnapshot()
    expect(component.update()).toMatchSnapshot()
  })
})
