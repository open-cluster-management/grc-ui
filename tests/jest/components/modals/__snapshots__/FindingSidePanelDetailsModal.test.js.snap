// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindingSidePanelDetailsModal renders as expected 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(Connect(FindingSidePanelDetailsModal))
    data={
      Object {
        "__typename": "Occurrence",
        "context": Object {
          "accountId": "id-mycluster-account",
          "clusterName": "clusterhub",
          "namespaceName": "Excludes: [kube-*], Includes: [default]",
          "region": "clusterhub",
          "resourceCrn": null,
          "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
          "resourceName": "policy-mutationpolicy",
          "resourceType": "Policy",
          "serviceCrn": null,
          "serviceName": "icp4mcm-findings",
        },
        "custom": Object {
          "context": "Policy: policy-mutationpolicy",
          "finding.severity": "Medium",
          "securityClassification.securityCategories": "System And Information Integrity",
          "securityClassification.securityControl": "Mutation Advisor",
          "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
          "updateTime": "-",
        },
        "finding": Object {
          "certainty": "HIGH",
          "dataTransferred": null,
          "networkConnection": null,
          "nextSteps": Array [
            Object {
              "title": "View the details for the compliance problem in the occurrence of the findings.",
              "url": null,
            },
          ],
          "severity": "MEDIUM",
        },
        "kind": "HCMSecurityFindings",
        "longDescription": "MCM Policy that is not compliant",
        "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
        "providerId": "security-advisor",
        "providerName": "id-mycluster-account/providers/security-advisor",
        "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
        "reportedBy": Object {
          "id": "mcm-policy-adapter",
          "title": "Security Advisor MCM Policy Findings Adapter",
          "url": null,
        },
        "securityClassification": Object {
          "securityCategories": Array [
            "SystemAndInformationIntegrity",
          ],
          "securityControl": "MutationAdvisor",
          "securityStandards": Array [
            "HIPAA",
            " NIST",
            " PCI",
            " FISMA",
          ],
        },
        "shortDescription": "Policy that is not compliant",
        "updateTime": "2019-10-07T16:33:50.061244Z",
      }
    }
    history={Object {}}
    locale="en"
    location={
      Object {
        "ancestorOrigins": Object {},
        "hash": "",
        "host": "localhost:3000",
        "hostname": "localhost",
        "href": "https://localhost:3000/multicloud/policies/findings",
        "origin": "https://localhost:3000",
        "pathname": "/multicloud/policies/findings",
        "port": "3000",
        "protocol": "https:",
        "search": "",
      }
    }
    open={true}
    resourceType={
      Object {
        "list": "HCMSecurityFindingsList",
        "name": "HCMSecurityFindings",
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    title="policy that is not compliant"
    updateModal={[MockFunction]}
  />
</Router>
`;

exports[`FindingSidePanelDetailsModal renders as expected 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Connect(FindingSidePanelDetailsModal))
      data={
        Object {
          "__typename": "Occurrence",
          "context": Object {
            "accountId": "id-mycluster-account",
            "clusterName": "clusterhub",
            "namespaceName": "Excludes: [kube-*], Includes: [default]",
            "region": "clusterhub",
            "resourceCrn": null,
            "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
            "resourceName": "policy-mutationpolicy",
            "resourceType": "Policy",
            "serviceCrn": null,
            "serviceName": "icp4mcm-findings",
          },
          "custom": Object {
            "context": "Policy: policy-mutationpolicy",
            "finding.severity": "Medium",
            "securityClassification.securityCategories": "System And Information Integrity",
            "securityClassification.securityControl": "Mutation Advisor",
            "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
            "updateTime": "-",
          },
          "finding": Object {
            "certainty": "HIGH",
            "dataTransferred": null,
            "networkConnection": null,
            "nextSteps": Array [
              Object {
                "title": "View the details for the compliance problem in the occurrence of the findings.",
                "url": null,
              },
            ],
            "severity": "MEDIUM",
          },
          "kind": "HCMSecurityFindings",
          "longDescription": "MCM Policy that is not compliant",
          "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
          "providerId": "security-advisor",
          "providerName": "id-mycluster-account/providers/security-advisor",
          "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
          "reportedBy": Object {
            "id": "mcm-policy-adapter",
            "title": "Security Advisor MCM Policy Findings Adapter",
            "url": null,
          },
          "securityClassification": Object {
            "securityCategories": Array [
              "SystemAndInformationIntegrity",
            ],
            "securityControl": "MutationAdvisor",
            "securityStandards": Array [
              "HIPAA",
              " NIST",
              " PCI",
              " FISMA",
            ],
          },
          "shortDescription": "Policy that is not compliant",
          "updateTime": "2019-10-07T16:33:50.061244Z",
        }
      }
      history={Object {}}
      locale="en"
      location={
        Object {
          "ancestorOrigins": Object {},
          "hash": "",
          "host": "localhost:3000",
          "hostname": "localhost",
          "href": "https://localhost:3000/multicloud/policies/findings",
          "origin": "https://localhost:3000",
          "pathname": "/multicloud/policies/findings",
          "port": "3000",
          "protocol": "https:",
          "search": "",
        }
      }
      open={true}
      resourceType={
        Object {
          "list": "HCMSecurityFindingsList",
          "name": "HCMSecurityFindings",
        }
      }
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      title="policy that is not compliant"
      updateModal={[MockFunction]}
    />
  </Router>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "data": Object {
        "__typename": "Occurrence",
        "context": Object {
          "accountId": "id-mycluster-account",
          "clusterName": "clusterhub",
          "namespaceName": "Excludes: [kube-*], Includes: [default]",
          "region": "clusterhub",
          "resourceCrn": null,
          "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
          "resourceName": "policy-mutationpolicy",
          "resourceType": "Policy",
          "serviceCrn": null,
          "serviceName": "icp4mcm-findings",
        },
        "custom": Object {
          "context": "Policy: policy-mutationpolicy",
          "finding.severity": "Medium",
          "securityClassification.securityCategories": "System And Information Integrity",
          "securityClassification.securityControl": "Mutation Advisor",
          "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
          "updateTime": "-",
        },
        "finding": Object {
          "certainty": "HIGH",
          "dataTransferred": null,
          "networkConnection": null,
          "nextSteps": Array [
            Object {
              "title": "View the details for the compliance problem in the occurrence of the findings.",
              "url": null,
            },
          ],
          "severity": "MEDIUM",
        },
        "kind": "HCMSecurityFindings",
        "longDescription": "MCM Policy that is not compliant",
        "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
        "providerId": "security-advisor",
        "providerName": "id-mycluster-account/providers/security-advisor",
        "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
        "reportedBy": Object {
          "id": "mcm-policy-adapter",
          "title": "Security Advisor MCM Policy Findings Adapter",
          "url": null,
        },
        "securityClassification": Object {
          "securityCategories": Array [
            "SystemAndInformationIntegrity",
          ],
          "securityControl": "MutationAdvisor",
          "securityStandards": Array [
            "HIPAA",
            " NIST",
            " PCI",
            " FISMA",
          ],
        },
        "shortDescription": "Policy that is not compliant",
        "updateTime": "2019-10-07T16:33:50.061244Z",
      },
      "history": Object {},
      "locale": "en",
      "location": Object {
        "ancestorOrigins": Object {},
        "hash": "",
        "host": "localhost:3000",
        "hostname": "localhost",
        "href": "https://localhost:3000/multicloud/policies/findings",
        "origin": "https://localhost:3000",
        "pathname": "/multicloud/policies/findings",
        "port": "3000",
        "protocol": "https:",
        "search": "",
      },
      "open": true,
      "resourceType": Object {
        "list": "HCMSecurityFindingsList",
        "name": "HCMSecurityFindings",
      },
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "title": "policy that is not compliant",
      "updateModal": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "__typename": "Occurrence",
          "context": Object {
            "accountId": "id-mycluster-account",
            "clusterName": "clusterhub",
            "namespaceName": "Excludes: [kube-*], Includes: [default]",
            "region": "clusterhub",
            "resourceCrn": null,
            "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
            "resourceName": "policy-mutationpolicy",
            "resourceType": "Policy",
            "serviceCrn": null,
            "serviceName": "icp4mcm-findings",
          },
          "custom": Object {
            "context": "Policy: policy-mutationpolicy",
            "finding.severity": "Medium",
            "securityClassification.securityCategories": "System And Information Integrity",
            "securityClassification.securityControl": "Mutation Advisor",
            "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
            "updateTime": "-",
          },
          "finding": Object {
            "certainty": "HIGH",
            "dataTransferred": null,
            "networkConnection": null,
            "nextSteps": Array [
              Object {
                "title": "View the details for the compliance problem in the occurrence of the findings.",
                "url": null,
              },
            ],
            "severity": "MEDIUM",
          },
          "kind": "HCMSecurityFindings",
          "longDescription": "MCM Policy that is not compliant",
          "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
          "providerId": "security-advisor",
          "providerName": "id-mycluster-account/providers/security-advisor",
          "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
          "reportedBy": Object {
            "id": "mcm-policy-adapter",
            "title": "Security Advisor MCM Policy Findings Adapter",
            "url": null,
          },
          "securityClassification": Object {
            "securityCategories": Array [
              "SystemAndInformationIntegrity",
            ],
            "securityControl": "MutationAdvisor",
            "securityStandards": Array [
              "HIPAA",
              " NIST",
              " PCI",
              " FISMA",
            ],
          },
          "shortDescription": "Policy that is not compliant",
          "updateTime": "2019-10-07T16:33:50.061244Z",
        },
        "history": Object {},
        "locale": "en",
        "location": Object {
          "ancestorOrigins": Object {},
          "hash": "",
          "host": "localhost:3000",
          "hostname": "localhost",
          "href": "https://localhost:3000/multicloud/policies/findings",
          "origin": "https://localhost:3000",
          "pathname": "/multicloud/policies/findings",
          "port": "3000",
          "protocol": "https:",
          "search": "",
        },
        "open": true,
        "resourceType": Object {
          "list": "HCMSecurityFindingsList",
          "name": "HCMSecurityFindings",
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "title": "policy that is not compliant",
        "updateModal": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "static getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`FindingSidePanelDetailsModal renders as expected 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <withRouter(Connect(FindingSidePanelDetailsModal))
      data={
        Object {
          "__typename": "Occurrence",
          "context": Object {
            "accountId": "id-mycluster-account",
            "clusterName": "clusterhub",
            "namespaceName": "Excludes: [kube-*], Includes: [default]",
            "region": "clusterhub",
            "resourceCrn": null,
            "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
            "resourceName": "policy-mutationpolicy",
            "resourceType": "Policy",
            "serviceCrn": null,
            "serviceName": "icp4mcm-findings",
          },
          "custom": Object {
            "context": "Policy: policy-mutationpolicy",
            "finding.severity": "Medium",
            "securityClassification.securityCategories": "System And Information Integrity",
            "securityClassification.securityControl": "Mutation Advisor",
            "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
            "updateTime": "-",
          },
          "finding": Object {
            "certainty": "HIGH",
            "dataTransferred": null,
            "networkConnection": null,
            "nextSteps": Array [
              Object {
                "title": "View the details for the compliance problem in the occurrence of the findings.",
                "url": null,
              },
            ],
            "severity": "MEDIUM",
          },
          "kind": "HCMSecurityFindings",
          "longDescription": "MCM Policy that is not compliant",
          "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
          "providerId": "security-advisor",
          "providerName": "id-mycluster-account/providers/security-advisor",
          "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
          "reportedBy": Object {
            "id": "mcm-policy-adapter",
            "title": "Security Advisor MCM Policy Findings Adapter",
            "url": null,
          },
          "securityClassification": Object {
            "securityCategories": Array [
              "SystemAndInformationIntegrity",
            ],
            "securityControl": "MutationAdvisor",
            "securityStandards": Array [
              "HIPAA",
              " NIST",
              " PCI",
              " FISMA",
            ],
          },
          "shortDescription": "Policy that is not compliant",
          "updateTime": "2019-10-07T16:33:50.061244Z",
        }
      }
      history={Object {}}
      locale="en"
      location={
        Object {
          "ancestorOrigins": Object {},
          "hash": "",
          "host": "localhost:3000",
          "hostname": "localhost",
          "href": "https://localhost:3000/multicloud/policies/findings",
          "origin": "https://localhost:3000",
          "pathname": "/multicloud/policies/findings",
          "port": "3000",
          "protocol": "https:",
          "search": "",
        }
      }
      open={true}
      resourceType={
        Object {
          "list": "HCMSecurityFindingsList",
          "name": "HCMSecurityFindings",
        }
      }
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      title="policy that is not compliant"
      updateModal={[MockFunction]}
    />
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <withRouter(Connect(FindingSidePanelDetailsModal))
        data={
          Object {
            "__typename": "Occurrence",
            "context": Object {
              "accountId": "id-mycluster-account",
              "clusterName": "clusterhub",
              "namespaceName": "Excludes: [kube-*], Includes: [default]",
              "region": "clusterhub",
              "resourceCrn": null,
              "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
              "resourceName": "policy-mutationpolicy",
              "resourceType": "Policy",
              "serviceCrn": null,
              "serviceName": "icp4mcm-findings",
            },
            "custom": Object {
              "context": "Policy: policy-mutationpolicy",
              "finding.severity": "Medium",
              "securityClassification.securityCategories": "System And Information Integrity",
              "securityClassification.securityControl": "Mutation Advisor",
              "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
              "updateTime": "-",
            },
            "finding": Object {
              "certainty": "HIGH",
              "dataTransferred": null,
              "networkConnection": null,
              "nextSteps": Array [
                Object {
                  "title": "View the details for the compliance problem in the occurrence of the findings.",
                  "url": null,
                },
              ],
              "severity": "MEDIUM",
            },
            "kind": "HCMSecurityFindings",
            "longDescription": "MCM Policy that is not compliant",
            "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
            "providerId": "security-advisor",
            "providerName": "id-mycluster-account/providers/security-advisor",
            "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
            "reportedBy": Object {
              "id": "mcm-policy-adapter",
              "title": "Security Advisor MCM Policy Findings Adapter",
              "url": null,
            },
            "securityClassification": Object {
              "securityCategories": Array [
                "SystemAndInformationIntegrity",
              ],
              "securityControl": "MutationAdvisor",
              "securityStandards": Array [
                "HIPAA",
                " NIST",
                " PCI",
                " FISMA",
              ],
            },
            "shortDescription": "Policy that is not compliant",
            "updateTime": "2019-10-07T16:33:50.061244Z",
          }
        }
        history={Object {}}
        locale="en"
        location={
          Object {
            "ancestorOrigins": Object {},
            "hash": "",
            "host": "localhost:3000",
            "hostname": "localhost",
            "href": "https://localhost:3000/multicloud/policies/findings",
            "origin": "https://localhost:3000",
            "pathname": "/multicloud/policies/findings",
            "port": "3000",
            "protocol": "https:",
            "search": "",
          }
        }
        open={true}
        resourceType={
          Object {
            "list": "HCMSecurityFindingsList",
            "name": "HCMSecurityFindings",
          }
        }
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        title="policy that is not compliant"
        updateModal={[MockFunction]}
      />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "__typename": "Occurrence",
          "context": Object {
            "accountId": "id-mycluster-account",
            "clusterName": "clusterhub",
            "namespaceName": "Excludes: [kube-*], Includes: [default]",
            "region": "clusterhub",
            "resourceCrn": null,
            "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
            "resourceName": "policy-mutationpolicy",
            "resourceType": "Policy",
            "serviceCrn": null,
            "serviceName": "icp4mcm-findings",
          },
          "custom": Object {
            "context": "Policy: policy-mutationpolicy",
            "finding.severity": "Medium",
            "securityClassification.securityCategories": "System And Information Integrity",
            "securityClassification.securityControl": "Mutation Advisor",
            "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
            "updateTime": "-",
          },
          "finding": Object {
            "certainty": "HIGH",
            "dataTransferred": null,
            "networkConnection": null,
            "nextSteps": Array [
              Object {
                "title": "View the details for the compliance problem in the occurrence of the findings.",
                "url": null,
              },
            ],
            "severity": "MEDIUM",
          },
          "kind": "HCMSecurityFindings",
          "longDescription": "MCM Policy that is not compliant",
          "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
          "providerId": "security-advisor",
          "providerName": "id-mycluster-account/providers/security-advisor",
          "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
          "reportedBy": Object {
            "id": "mcm-policy-adapter",
            "title": "Security Advisor MCM Policy Findings Adapter",
            "url": null,
          },
          "securityClassification": Object {
            "securityCategories": Array [
              "SystemAndInformationIntegrity",
            ],
            "securityControl": "MutationAdvisor",
            "securityStandards": Array [
              "HIPAA",
              " NIST",
              " PCI",
              " FISMA",
            ],
          },
          "shortDescription": "Policy that is not compliant",
          "updateTime": "2019-10-07T16:33:50.061244Z",
        },
        "history": Object {},
        "locale": "en",
        "location": Object {
          "ancestorOrigins": Object {},
          "hash": "",
          "host": "localhost:3000",
          "hostname": "localhost",
          "href": "https://localhost:3000/multicloud/policies/findings",
          "origin": "https://localhost:3000",
          "pathname": "/multicloud/policies/findings",
          "port": "3000",
          "protocol": "https:",
          "search": "",
        },
        "open": true,
        "resourceType": Object {
          "list": "HCMSecurityFindingsList",
          "name": "HCMSecurityFindings",
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "title": "policy that is not compliant",
        "updateModal": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withRouter(Connect(FindingSidePanelDetailsModal))
          data={
            Object {
              "__typename": "Occurrence",
              "context": Object {
                "accountId": "id-mycluster-account",
                "clusterName": "clusterhub",
                "namespaceName": "Excludes: [kube-*], Includes: [default]",
                "region": "clusterhub",
                "resourceCrn": null,
                "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
                "resourceName": "policy-mutationpolicy",
                "resourceType": "Policy",
                "serviceCrn": null,
                "serviceName": "icp4mcm-findings",
              },
              "custom": Object {
                "context": "Policy: policy-mutationpolicy",
                "finding.severity": "Medium",
                "securityClassification.securityCategories": "System And Information Integrity",
                "securityClassification.securityControl": "Mutation Advisor",
                "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
                "updateTime": "-",
              },
              "finding": Object {
                "certainty": "HIGH",
                "dataTransferred": null,
                "networkConnection": null,
                "nextSteps": Array [
                  Object {
                    "title": "View the details for the compliance problem in the occurrence of the findings.",
                    "url": null,
                  },
                ],
                "severity": "MEDIUM",
              },
              "kind": "HCMSecurityFindings",
              "longDescription": "MCM Policy that is not compliant",
              "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
              "providerId": "security-advisor",
              "providerName": "id-mycluster-account/providers/security-advisor",
              "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
              "reportedBy": Object {
                "id": "mcm-policy-adapter",
                "title": "Security Advisor MCM Policy Findings Adapter",
                "url": null,
              },
              "securityClassification": Object {
                "securityCategories": Array [
                  "SystemAndInformationIntegrity",
                ],
                "securityControl": "MutationAdvisor",
                "securityStandards": Array [
                  "HIPAA",
                  " NIST",
                  " PCI",
                  " FISMA",
                ],
              },
              "shortDescription": "Policy that is not compliant",
              "updateTime": "2019-10-07T16:33:50.061244Z",
            }
          }
          history={Object {}}
          locale="en"
          location={
            Object {
              "ancestorOrigins": Object {},
              "hash": "",
              "host": "localhost:3000",
              "hostname": "localhost",
              "href": "https://localhost:3000/multicloud/policies/findings",
              "origin": "https://localhost:3000",
              "pathname": "/multicloud/policies/findings",
              "port": "3000",
              "protocol": "https:",
              "search": "",
            }
          }
          open={true}
          resourceType={
            Object {
              "list": "HCMSecurityFindingsList",
              "name": "HCMSecurityFindings",
            }
          }
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          title="policy that is not compliant"
          updateModal={[MockFunction]}
        />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "__typename": "Occurrence",
            "context": Object {
              "accountId": "id-mycluster-account",
              "clusterName": "clusterhub",
              "namespaceName": "Excludes: [kube-*], Includes: [default]",
              "region": "clusterhub",
              "resourceCrn": null,
              "resourceId": "bf080cda-e44a-11e9-8895-005056a061f1",
              "resourceName": "policy-mutationpolicy",
              "resourceType": "Policy",
              "serviceCrn": null,
              "serviceName": "icp4mcm-findings",
            },
            "custom": Object {
              "context": "Policy: policy-mutationpolicy",
              "finding.severity": "Medium",
              "securityClassification.securityCategories": "System And Information Integrity",
              "securityClassification.securityControl": "Mutation Advisor",
              "securityClassification.securityStandards": "HIPAA, NIST, PCI, FISMA",
              "updateTime": "-",
            },
            "finding": Object {
              "certainty": "HIGH",
              "dataTransferred": null,
              "networkConnection": null,
              "nextSteps": Array [
                Object {
                  "title": "View the details for the compliance problem in the occurrence of the findings.",
                  "url": null,
                },
              ],
              "severity": "MEDIUM",
            },
            "kind": "HCMSecurityFindings",
            "longDescription": "MCM Policy that is not compliant",
            "name": "id-mycluster-account/providers/security-advisor/occurrences/clusterhub-policy-bf080cda-e44a-11e9-8895-005056a061f1",
            "providerId": "security-advisor",
            "providerName": "id-mycluster-account/providers/security-advisor",
            "remediation": "mapping error from raw object: \`no matches for kind \\"MutationPolicy\\" in version \\"policies.ibm.com/v1alpha1\\"\`",
            "reportedBy": Object {
              "id": "mcm-policy-adapter",
              "title": "Security Advisor MCM Policy Findings Adapter",
              "url": null,
            },
            "securityClassification": Object {
              "securityCategories": Array [
                "SystemAndInformationIntegrity",
              ],
              "securityControl": "MutationAdvisor",
              "securityStandards": Array [
                "HIPAA",
                " NIST",
                " PCI",
                " FISMA",
              ],
            },
            "shortDescription": "Policy that is not compliant",
            "updateTime": "2019-10-07T16:33:50.061244Z",
          },
          "history": Object {},
          "locale": "en",
          "location": Object {
            "ancestorOrigins": Object {},
            "hash": "",
            "host": "localhost:3000",
            "hostname": "localhost",
            "href": "https://localhost:3000/multicloud/policies/findings",
            "origin": "https://localhost:3000",
            "pathname": "/multicloud/policies/findings",
            "port": "3000",
            "protocol": "https:",
            "search": "",
          },
          "open": true,
          "resourceType": Object {
            "list": "HCMSecurityFindingsList",
            "name": "HCMSecurityFindings",
          },
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "title": "policy that is not compliant",
          "updateModal": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "static getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
